[Cong Le] Implement add competency

ltc2310 opened the pull request.

conglet2310 committed:
[CF] complete disable and enable compentency

conglet2310 committed:
[CF] Add new competency and update activate competency.
Add new competency : It doesn't update page in real time
Update activate competency : It works but don't update page and onChange function when I click in the first time it doesn't work,on second time it works but the value opposite with data in firebase

conglet2310 committed:
[CF] Add new competency and update activate competency.
Add new competency : It doesn't update page in real time
Update activate competency : It works but don't update page and onChange function when I click in the first time it doesn't work,on second time it works but the value opposite with data in firebase

ltc2310 commented:
[CF] Add new competency and update activate competency.
Add new competency : It doesn't update page in real time
Update activate competency : It works but don't update page and onChange function when I click in the first time it doesn't work,on second time it works but the value opposite with data in firebase

conglet2310 committed:
[CF] create competency question

hunterbmt commented on src/Competencies.js:
```diff
@@ -0,0 +1,293 @@
+import './competency.css';
+import {Router, Link} from 'react-router-component';
+
+var list;
```
Do not use "var" in ES6

hunterbmt commented on src/Competencies.js:
```diff
@@ -0,0 +1,293 @@
+class CompetencyList extends Component {
+  constructor(props) {
+     super(props);
+     list=this;
```
Why do we need to assign this to outside variable?

hunterbmt commented on src/Competencies.js:
```diff
@@ -0,0 +1,293 @@
+    }
+}
+
+class CompetencyPopup extends Component {
```
one component per file

hunterbmt commented on src/Competencies.js:
```diff
@@ -0,0 +1,293 @@
+      visible: false
+    });
+
+    console.log(list.state.arrayCompetencies);
```
Removing console.log

hunterbmt commented on src/Competencies.js:
```diff
@@ -0,0 +1,293 @@
+
+    this.addNewCompetency(this.state.competencyName);
+
+    var newArrayCompetencies = list.state.arrayCompetencies.push(this.state.competencyName);
```
Use this instead of outside variable

conglet2310 committed:
[CF] fix Competencies

conglet2310 committed:
[CF] fix comment and rewrite function follow new database

conglet2310 committed:
[CF] Add Competency
Fix comment and rewrite function follow new database
How to set auto increment key on competencies when I save new competency?
When I get data source for competencies KMScore to show for user,It return double data competency?

ltc2310 commented:
Fix comment and rewrite function follow new database
How to set auto increment key on competencies when I save new competency?
When I get data source for competencies KMScore to show on table,It return double data competency show on table?

conglet2310 committed:
[CFT] fix add competency follow new database but it has some issues
- I can't set auto increment when I save new competency
- I can't load two table KMS Core and KMS optional in the same time
- When I click on button it always append data on the table

ltc2310 commented:
* [CFT] fix add competency follow new database but it has some issues
- I can't set auto increment when I save new competency
- I can't load two table KMS Core and KMS optional in the same time
- When I click on button it always append data on the table

hunterbmt commented on src/Competencies.js:
```diff
@@ -0,0 +1,314 @@
+  }
+
+  getDataSoureKMSCore() {
+    _.map(this.state.competenciesKMSCore, (item, index) => {
```
const dataSourceCore = _.map(() => return {}) ;
this.setState({dataSourceCode});

conglet2310 committed:
[CFT] add competency
- Complete all features but don't validate data
- I have to reload page if i create or update competency

conglet2310 committed:
* [CFT] add competency
- Complete all features but don't validate data
- I have to reload page if i create or update competency

conglet2310 committed:
[CFT] Add Competency auto increment Id and fix UI

conglet2310 committed:
[CFT] add link for user add question

conglet2310 committed:
[CFT] Complete add competencies but have to reload page to see change

conglet2310 committed:
[CF] fix Add competency
add default index when add first competency
